lib/makwa/interaction.rb

```diff
@@ -2,6 +2,16 @@
 
 module Makwa
   class Interaction < ::ActiveInteraction::Base
+    class Interrupt < StandardError
+      attr_reader :errors
+
+      def initialize(errors)
+        super
+
+        @errors = errors
+      end
+    end
+
```

Comment:
I agree, that the rescue statement is ugly, however, I prefer the ugly method call over re-implementing ActiveInteractions' #run method.

How about we define our own interrupt class in the following way, replacing your implementation:

```
    class Interrupt < Object.const_get("::ActiveInteraction::Interrupt")
    end
```

And then we don't have to re-implement ActiveInteraction's #run method. We are moving the ugliness to a different place so that the rescue looks a bit cleaner.

Reply:
Brilliant!